src/libraries/System.Private.CoreLib/src/System/Char.cs

```diff
@@ -274,7 +274,7 @@ internal static bool TryParse(ReadOnlySpan<char> s, out char result)
         /// This determines whether the character is in the range 'A' through 'Z', inclusive,
         /// 'a' through 'z', inclusive, or '0' through '9', inclusive.
         /// </remarks>
-        public static bool IsAsciiLetterOrDigit(char c) => IsAsciiLetter(c) | IsBetween(c, '0', '9');
+        public static bool IsAsciiLetterOrDigit(char c) => IsAsciiLetter(c) | IsBetween(c, '0', '9');  // Using | here is an optimization (avoids branching).
```

Comment:
Thanks. There are a variety of places we use `|` like this, which is part of its raison d'etre. Why without a comment is there an assumption that it's a mistake and why is this particular use special so as to warrant a dedicated comment?

Reply:
When I do a quick search through this repo for `" | "`, I don't see many instances of `|` being used for boolean operations.

If there were hundreds (or even dozens) of similar usage patterns, I'd agree that it's a common-enough practice as to not warrant any clarification.

It seems, however, there are only a few.

The net effect is that the codebase exhibits these occasional inconsistencies.  Mentally, when you encounter one, it gives you pause:  *"Wait.  That doesn't look right. 
 Why's that?"*

Maybe you remember why it's that way.  Maybe you have to jog your memory.  Maybe you're stumped.  (Most people would be stumped.)

Either way, you've wasted time contemplating it and you're likely to do it all over again when you re-encounter it at some in the future.

A simple comment here short-circuits (pun intended) that whole mental exercise.

If there are indeed only a few such instances within this repo, I'd be happy to amend this PR to cover those as well.

Reply:
Thanks. Personally, I still don't believe this is valuable. The only reason to use a | here is for the exact reason the comment would state.

Reply:
My guess is that most C# programmers aren't even aware that this is valid syntax for a `boolean` expression.

If you put yourself in their 👞👞 ...?